### PR TITLE
Post like, bookmark and follow

### PR DESCRIPTION
**Pull request by tarunsankhla**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -62,7 +66,15 @@ function App() {
               </Suspense>
             </RequiredAuth>}
           />
+          <Route path={ROUTES.ROUTE_PATH_POSTPAGE} element={
+            <RequiredAuth>
+              <Suspense fallback={"..loading"}>
+                <PostPage />
```
Comment (m26unkwn) [inline src/App.js:72]: write fallback component it will be good ui and experience

**Review by m26unkwn — APPROVED** [review]
Try to extract login from JSX.

don't write it directly inside of jsx

File: src/App.css [hunk 2]
```diff
@@ -169,4 +173,12 @@ span {
...
 .fn-wg-600{
 	font-weight: 600;
+}
+
+.handle-img-np{
+    height: 2.5em;
+    width: 2.5em;
+    border-radius: 50%;
+    background-color: var(--primary-color);
+    color: white;
```
Comment (pankaj-prog) [inline src/App.css]: use color variables

File: src/App.js [hunk 3]
```diff
@@ -62,7 +66,15 @@ function App() {
...
+              </Suspense>
+            </RequiredAuth>}
+          />
         </Route>
+        
         {/* <Route path={ROUTES.ROUTE_PATH_MOCKMAN} element={<div className='MockAPI'><Mockman /></div>} /> */}
```
Comment (pankaj-prog) [inline src/App.js]: remove comment

File: src/components/common/BesideNavbar/BesideNavbar.css [hunk 4]
```diff
@@ -0,0 +1,76 @@
+.BesideNavbar{
+    width:35vw;
+    display: flex;
+    background-color: var(--black-hl-p5);
+    display: flex;
```
Comment (pankaj-prog) [inline src/components/common/BesideNavbar/BesideNavbar.css]: you are repeating display flex here

File: src/components/common/BesideNavbar/BesideNavbar.css [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    padding: 10px;
+    position: sticky;
+    top: 0;
+}
+
+
+.beside-navbar-container{
+    display: flex;
+    justify-content: flex-start;
```
Comment (pankaj-prog) [inline src/components/common/BesideNavbar/BesideNavbar.css]: this is not needed, as this is the default property

File: src/components/common/BesideNavbar/BesideNavbar.jsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                <div className='beside-container-header'>
+                    <span className='fn-wg-700'>Who to Follow?</span>
+                    <span className='fn-wg-700 red-txt'>Show More</span>
+                </div>
+                <hr/>
+                <div className='beside-user-data-container'>
+                    {
+                    userFollowArray.map(user => (
+                        <div className='user-data-beside-section' key={user.userId}>
+                            <div className='user-data-beside-section-name'>
```
Comment (pankaj-prog) [inline src/components/common/BesideNavbar/BesideNavbar.jsx]: use semantic tags, good for accessibility as well as SEO

File: src/components/common/Posts/Post.jsx [hunk 7]
```diff
@@ -1,39 +1,222 @@
...
+import {doc, getDoc, updateDoc} from 'firebase/firestore';
+import { useUserData } from 'context/UserContext';
+import { GetIndividualUserData } from 'utils/authService';
+import { Link, NavLink } from 'react-router-dom';
 
 const Post = ({props}) => {
-    console.log(props)
+    const { userState, userDispatch } = useAuth();
+    const { userData, setUserData } = useUserData();
+    // console.log(props)
```
Comment (pankaj-prog) [inline src/components/common/Posts/Post.jsx]: remove comment

File: src/utils/authService.js [hunk 8]
```diff
@@ -26,13 +28,14 @@ const LoginInWithEmail = async (data, userDispatch, navigate) => {
...
-      userId: response?.user?.uid ?? "",
-      photo: response.user.photoURL ?? "",
-    }
-    CreateUser(obj);
+    // let obj = {
+    //   name: response?.user?.displayName ?? "",
+    //   emailId: response?.user?.email ?? "",
+    //   userId: response?.user?.uid ?? "",
+    //   photo: response.user.photoURL ?? "",
+    // }
```
Comment (pankaj-prog) [inline src/utils/authService.js]: remove these comments

File: src/App.js (hunk 1 above)
Comment (tarunsankhla) [reply src/App.js:72]: will do that once bare minimum is done with all p0

File: src/App.css (hunk 2 above)
Comment (tarunsankhla) [reply src/App.css]: i guess this are already variable provide by CSS i guess creating variable for default color is redoing stuff

File: src/App.js (hunk 3 above)
Comment (tarunsankhla) [reply src/App.js]: done

File: src/components/common/BesideNavbar/BesideNavbar.css (hunk 4 above)
Comment (tarunsankhla) [reply src/components/common/BesideNavbar/BesideNavbar.css]: done

File: src/components/common/BesideNavbar/BesideNavbar.css (hunk 5 above)
Comment (tarunsankhla) [reply src/components/common/BesideNavbar/BesideNavbar.css]: oh got it

File: src/components/common/BesideNavbar/BesideNavbar.jsx (hunk 6 above)
Comment (tarunsankhla) [reply src/components/common/BesideNavbar/BesideNavbar.jsx]: okay

File: src/utils/authService.js (hunk 8 above)
Comment (tarunsankhla) [reply src/utils/authService.js]: done

File: src/components/common/Posts/Post.jsx (hunk 7 above)
Comment (tarunsankhla) [reply src/components/common/Posts/Post.jsx]: done

**Pull request merged**